sirepo/job_driver/sbatch.py

```diff
@@ -112,10 +113,11 @@ def _agent_env(self):
                 script = f'''#!/bin/bash
 {self._agent_start_dev()}
 set -e
+pkill -u $USER -f 'sirepo job_agent start_sbatch' >& /dev/null || true
```

Comment:
I'm a bit tempted to be a bit more robust here. There's a problem if we are running alpha, beta, and prod on the same host (Cori). pkill was a quick hack. We should use a pid file.  Remove the pkill and after the `cd` do something like:

```sh
if [[ -r job_agent.pid ]]; then
     x=( $(<job_agent.pid) )
     if [[ ${x[0]} == $(hostname -f) ]]; then
          kill -KILL "${x[1]}" >& /dev/null || true
     else
          # you can ssh to a specific login node at NERSC s
          ssh "${x[0]}" kill -KILL "${x[1]}" >& /dev/null || true
     fi
     rm  job_agent.pid
fi
... start the process ...
echo "$(hostname -f) $!" > job_agent.pid
```

There are a number of questions here that might be better resolved by having the job_agent.py do all of the above. For example, how do we handle locking, or is that really the supervisor's responsibility? Should local agent do the same thing? Docker is different, of course.

Reply:
Went with the job_agent.py route. Agent now writes pid file and kills existing agent on startup.

@robnagler on NERSC how is SSH auth handled from one login node to another? Keys or does the user have to supply their creds?

Reply:
It just works without a password. I'm guessing that the login nodes trust each other using identd or something.